protocol/x/subaccounts/keeper/isolated_subaccount.go

```diff
@@ -2,16 +2,49 @@ package keeper
 
 import (
 	"math"
+	"math/big"
 
 	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	assettypes "github.com/dydxprotocol/v4-chain/protocol/x/assets/types"
 	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
 	"github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 )
 
+type positionStateTransition uint
+
+const (
+	opened positionStateTransition = iota
+	closed
+)
+
+var positionStateTransitionStringMap = map[positionStateTransition]string{
+	opened: "opened",
+	closed: "closed",
+}
+
+func (t positionStateTransition) String() string {
+	result, exists := positionStateTransitionStringMap[t]
+	if !exists {
+		return "UnexpectedStateTransitionError"
+	}
+
+	return result
+}
+
+// Represents a state transition for an isolated perpetual.
+type isolatedPerpetualStateTransition struct {
+	perpetualId uint32
+	// TODO(DEC-715): Support non-USDC assets.
+	// USDC position size of the subaccount that has a state change for an isolated perpetual.
+	usdcQuantumsBeforeUpdate *big.Int
+	transition               positionStateTransition
```

Comment:
Can we add some documentation wha `transition` means? 
Related, does `isolatedPerpetualStateTransition` only make sense when a position goes from `open` to `closed`, or a new position is `opened`? Does it not apply to position size change?

Reply:
Added comment.
>Related, does isolatedPerpetualStateTransition only make sense when a position goes from open to closed, or a new position is opened? Does it not apply to position size change?

That's correct, it only makes sense when a position goes from open to closed or when a new position is opened. All the collateral backing an existing position will already have been moved into the isolated perpetual's collateral pool (due to the logic in this PR) when it's size is changed so there's no need to do any collateral transfers when the position size changes.